platform/viewer/src/lib/DICOMSR/dataExchange.js

```diff
@@ -0,0 +1,49 @@
+import OHIF from '@ohif/core';
+import { retrieveMeasurementFromSR, stowSRFromMeasurements } from './handleSR';
+import { getLatestSRSeries } from './srUtils';
+
+export const retrieveMeasurements = options => {
+  OHIF.log.info('[DICOMSR] retrieveMeasurements');
+
+  const { server } = options;
+  const latestSeries = getLatestSRSeries();
+
+  if (!latestSeries) return Promise.resolve({});
+
+  return retrieveMeasurementFromSR(latestSeries, server);
+};
+
+export const storeMeasurements = async (measurementData, filter, server) => {
+  OHIF.log.info('[DICOMSR] storeMeasurements');
+
+  if (!server || server.type !== 'dicomWeb') {
+    OHIF.log.error('[DICOMSR] DicomWeb server is required!');
+    return Promise.reject({});
+  }
+
+  const firstMeasurementKey = Object.keys(measurementData)[0];
+  const firstMeasurement = measurementData[firstMeasurementKey][0];
+  const studyInstanceUid =
+    firstMeasurement && firstMeasurement.studyInstanceUid;
+
+  try {
+    const { notSupportedTools } = await stowSRFromMeasurements(
+      measurementData,
+      server
+    );
+    if (studyInstanceUid) {
+      OHIF.studies.deleteStudyMetadataPromise(studyInstanceUid);
+    }
+    const message =
+      notSupportedTools.length > 0
+        ? 'Measurements were parcially saved, some of the tools are not supported.'
```

Comment:
@dannyrb , do you think its worth writing which tools aren't supported in the snackbar here? The list would currently be quite long and I'm not sure on the UX. Right now the user has no real information as to what was saved. Its a stop-gap until we have support for all core tools, but we will always need something like this for unsupported 3rd party annotation tools anyway.

They are currently logged to the console for devs, but I don't know how much we should expose to the user.

Reply:
Partially is spelled wrong.

Instead of checking measurements at time of save, can we highlight them in the measurements table or use the "warning" styling with a note at the top of the table that tools that look like X cannot be persisted at this time?

Notification could remain simple. Annotations saved/persisted successfully.

Reply:
I´ve reverted the snackbar messages, back to error/success and added the warings on measurements that can not be persisted at this point.

Reply:
![Screen Shot 2019-11-18 at 03 11 42](https://user-images.githubusercontent.com/2378326/69028579-2d2ce600-09b1-11ea-9631-0501d19d778d.png)

Reply:
I like this, looks a lot cleaner. @dannyrb does this PR get your green flag?

Reply:
Its minor, but it would be nice if the `Save measurements`button had some on hover/on click animation.

Reply:
I've forget to change that. 
Just updated the cursor and hover/action style.